Sources/CNIOWindows/include/CNIOWindows.h

```diff
@@ -20,6 +20,36 @@
 #include <WinSock2.h>
 #include <time.h>
 
+// This is a DDK type which is not available in the WinSDK as it is not part of
+// the shared, usermode (um), or ucrt portions of the code.  We must replicate
+// this datastructure manually from the MSDN references or the DDK.
+// https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/ntifs/ns-ntifs-_reparse_data_buffer
+typedef struct _REPARSE_DATA_BUFFER {
+  ULONG   ReparseTag;
+  USHORT  ReparseDataLength;
+  USHORT  Reserved;
+  union {
+    struct {
+      USHORT  SubstituteNameOffset;
+      USHORT  SubstituteNameLength;
+      USHORT  PrintNameOffset;
+      USHORT  PrintNameLength;
+      ULONG   Flags;
+      WCHAR   PathBuffer[1];
+    } SymbolicLinkReparseBuffer;
+    struct {
+      USHORT  SubstituteNameOffset;
+      USHORT  SubstituteNameLength;
+      USHORT  PrintNameOffset;
+      USHORT  PrintNameLength;
+      WCHAR   PathBuffer[1];
+    } MountPointReparseBuffer;
+    struct {
+      UCHAR   DataBuffer[1];
+    } GenericReparsaeBuffer;
+  } DUMMYUNIONNAME;
+} REPARSE_DATA_BUFFER, *PREPARSE_DATA_BUFFER;
```

Comment:
Is it worth us namespacing this struct? SwiftPM has an awkward habit of being really aggressive about including header files in unrelated places, and I worry that someone else may also have forward-declared this type.

Reply:
I don't think its worth namespacing really.  Ideally this would just be provided by the `import WinSDK` (or `#include <Windows.h>`) but its a km header rather than um.

Reply:
The reason to want to namespace it is that the bugs that manifest if this does go wrong are _extremely_ gnarly to debug.

Reply:
Namespacing it makes it harder for others to immediately connect what this is to the Microsoft side IMO.

Reply:
I don't really think so: adding `CNIOWindows_` is not likely to confuse, and [we have prior art](https://github.com/apple/swift-nio/blob/bbb0e92702e7ac549cb9a13ab86f96e83deaac63/Sources/CNIODarwin/include/CNIODarwin.h#L29-L32).

Reply:
I suppose, Im still not particularly fond of it - it does make some things more annoying.